### PR TITLE
feature: allow course instructors to remove pupils from course

### PR DESCRIPTION
**Pull request by dhenkel92**

### CODE REVIEW

**Comment by dhenkel92** [comment]
https://github.com/corona-school/project-user/issues/947

File: graphql/subcourse/mutations.ts [hunk 1]
```diff
@@ -250,20 +250,20 @@ export class MutateSubcourseResolver {
...
         @Arg('subcourseId') subcourseId: number,
         @Arg('pupilId', { nullable: true }) pupilId?: number
     ): Promise<boolean> {
-        const { user } = context;
-        const pupil = await getSessionPupil(context, pupilId);
-        const pupilUser = userForPupil(pupil);
         const subcourse = await getSubcourse(subcourseId);
         await hasAccess(context, 'Subcourse', subcourse);
 
+        const pupil = await getPupil(pupilId || context.user.pupilId);
```
Comment (Jonasdoubleyou) [inline graphql/subcourse/mutations.ts:262]: Participants could now throw out other users. pupilId may only be passed in by Owner and Admin.

**Review by Jonasdoubleyou — CHANGES_REQUESTED** [review]

File: graphql/subcourse/mutations.ts (hunk 1 above)
Comment (dhenkel92) [reply graphql/subcourse/mutations.ts:272]: Very good point. I've pushed an update.

File: graphql/subcourse/mutations.ts [hunk 2]
```diff
@@ -250,20 +250,30 @@ export class MutateSubcourseResolver {
...
     ): Promise<boolean> {
-        const { user } = context;
-        const pupil = await getSessionPupil(context, pupilId);
-        const pupilUser = userForPupil(pupil);
         const subcourse = await getSubcourse(subcourseId);
         await hasAccess(context, 'Subcourse', subcourse);
 
+        // Make sure that only Admins/Owners can remove other pupils from the course
+        if (context.user.pupilId && pupilId && context.user.pupilId !== pupilId) {
+            logger.warn(`User tried to remove other pupil from course`, {
```
Comment (Jonasdoubleyou) [inline graphql/subcourse/mutations.ts:264]: Could also just throw an AuthenticationError

**Review by Jonasdoubleyou — APPROVED** [review]

**Pull request merged**